src/main/java/dev/aziz/movies/services/MovieService.java

```diff
@@ -28,7 +37,53 @@ public MovieDto findMovieById(Long id) {
                 .stream()
                 .filter(movieDto -> movieDto.getId() == id)
                 .findFirst()
-                .orElseThrow();
+                .orElseThrow(() -> new AppException("Movie not found.", HttpStatus.NOT_FOUND));
         return foundMovie;
     }
+
+    public MovieDto createMovie(MovieDto movieDto) {
+        if (movieDtoList.contains(movieDto)) {
+            throw new AppException("Movie already exists.", HttpStatus.BAD_REQUEST);
+        }
+        MovieDto createdMovieDto = movieDto;
+        movieDtoList.add(movieDto);
+        return createdMovieDto;
+    }
+
+    public MovieDto deleteById(Long id) {
+        MovieDto movieById = findMovieById(id);
+        if (!movieDtoList.contains(movieById)) {
+            throw new AppException("Movie not found.", HttpStatus.NOT_FOUND);
+        }
+        movieDtoList.remove(movieById);
+        return movieById;
+    }
+
+    public MovieDto updateById(Long id, UpdateDto updateDto) {
+        MovieDto movieById = findMovieById(id);
+        if (!movieDtoList.contains(movieById)) {
+            throw new AppException("Movie not found.", HttpStatus.NOT_FOUND);
+        }
+
+        if ("".equals(updateDto.getDescription())){
```

Comment:
With the `@Valid` annotation in the controller, you don't need those validations in the service.